### PR TITLE
feat(postgres): add support for engine parameters

### PR DESCRIPTION
**Pull request by armoucar-neon**
# Description

- Introduced engine_params to support passing parameters to create_engine.
- Updated create_engine and create_async_engine calls to include engine_params.
- Initialized engine_params in the constructor.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [X] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [X] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py [hunk 1]
```diff
@@ -173,6 +173,7 @@ def __init__(
         debug: bool = False,
         use_jsonb: bool = False,
         hnsw_kwargs: Optional[Dict[str, Any]] = None,
+        engine_params: Optional[Dict[str, Any]] = None,
```
Comment (nerdai) [inline llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py:176]: nit: rename to `create_engine_kwargs`

**Review by nerdai — CHANGES_REQUESTED** [review]
looks good! thanks @armoucar-neon just a minor change requested. But please also bump the version number of the package in the `pyproject.toml`.

File: llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py (hunk 1 above)
Comment (armoucar-neon) [reply llama-index-integrations/vector_stores/llama-index-vector-stores-postgres/llama_index/vector_stores/postgres/base.py]: ✅

**Review by nerdai — APPROVED** [review]
thanks @armoucar-neon!

**Pull request merged**